compiler/rustc_typeck/src/check/cast.rs

```diff
@@ -92,7 +92,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let t = self.resolve_vars_if_possible(t);
 
         if t.references_error() {
-            return Err(ErrorReported);
+            return Err(ErrorReported::new_i_know_what_im_doing());
```

Comment:
Similar as elsewhere, `ErrorReported` should be available from `references_error` itself.